modules/github_integration/app/models/github_user.rb

```diff
@@ -0,0 +1,39 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+class GithubUser < ApplicationRecord
+  has_many :github_pull_requests
+  belongs_to :user, optional: true
```

Comment:
A User is somewhat independent of the GithubUser. But if the User gets deleted in OpenProject, the reference in here needs to be nullified. Deleting users has its own job (https://github.com/opf/openproject/blob/dev/app/workers/principals/delete_job.rb) where you might want to hook into

Reply:
Additionally, and I did not test this, but given that no assertions on the presence of the user exists I'd expect the application to be robust in case the user does not exist.

Reply:
to be honest, I guess I was too optimistic when I wrote the migration (secretly hoping that I can get github-user to op-user connections working). But since this is not in scope anymore and not functioning at the time of this PR, I'd rather remove the association.